lib/dal/src/workspace_snapshot/node_weight/component_node_weight.rs

```diff
@@ -488,9 +490,7 @@ impl CorrectTransforms for ComponentNodeWeight {
                     destination,
                 } if EdgeWeightKindDiscriminants::Use == edge_weight.kind().into()
                     && is_self(source)
-                    && graph
-                        .get_node_weight_by_id_opt(destination.id)
-                        .is_some_and(|n| NodeWeightDiscriminants::SchemaVariant == n.into()) =>
+                    && destination.node_weight_kind == NodeWeightDiscriminants::SchemaVariant =>
```

Comment:
This fixes a bug where we would not apply this correction in the case of net-new schema variants (it ensures the new variant does not have to be on the graph to apply the correction)